File: customer-address-jpa/src/main/java/com/izertis/example/domain/PaymentMethodType.java
package com.izertis.example.domain;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

import java.util.Arrays;

/** Enum for PaymentMethodType. */
public enum PaymentMethodType {

    VISA(1), MASTERCARD(2),
    ;

    private final Integer value;

    private PaymentMethodType(Integer value) {
        this.value = value;
    }

    public Integer getValue() {
        return value;
    }

    public static PaymentMethodType fromValue(Integer value) {
        return Arrays.stream(PaymentMethodType.values())
                .filter(e -> e.value.equals(value))
                .findFirst()
                .orElse(null);
    }

    @Converter
    static class PaymentMethodTypeConverter implements AttributeConverter<PaymentMethodType, Integer> {

        @Override
        public Integer convertToDatabaseColumn(PaymentMethodType attribute) {
            if (attribute == null) {
                return null;
            }

            return attribute.value;
        }

        @Override
        public PaymentMethodType convertToEntityAttribute(Integer dbData) {
            return PaymentMethodType.fromValue(dbData);
        }

    }

}

File: kustomer-address-jpa/src/main/kotlin/com/izertis/example/domain/PaymentMethodType.kt
package com.izertis.example.domain

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

/** Enum for PaymentMethodType. */
enum class PaymentMethodType(val value: Int) {

  VISA(1),
  MASTERCARD(2);

  companion object {
    fun fromValue(value: Int): PaymentMethodType? {
      return entries.find { it.value == value }
    }
  }

  @Converter
  class PaymentMethodTypeConverter : AttributeConverter<PaymentMethodType, Int> {
    override fun convertToDatabaseColumn(attribute: PaymentMethodType?): Int? {
      return attribute?.value
    }

    override fun convertToEntityAttribute(dbData: Int?): PaymentMethodType? {
      return if (dbData == null) null else PaymentMethodType.fromValue(dbData)
    }
  }
}
